tools/shared/src/main/scala/org/scalajs/core/tools/linker/Linker.scala

```diff
@@ -73,4 +73,77 @@ final class Linker(frontend: LinkerFrontend, backend: LinkerBackend)
   }
 }
 
-object Linker extends LinkerPlatformExtensions
+object Linker extends LinkerPlatformExtensions {
+  /** Configuration to be passed to the `apply()` method. */
+  final class Config private (
+      /** Whether to generate source maps. */
+      val sourceMap: Boolean,
+      /** Whether to use the Scala.js optimizer. */
+      val optimizer: Boolean,
+      /** Whether things that can be parallelized should be parallelized.
+       *  On the JavaScript platform, this does not have any effect.
+       */
+      val parallel: Boolean,
+      /** Whether to use the Google Closure Compiler pass.
+       *  On the JavaScript platform, this does not have any effect.
```

Comment:
We're not giving up type-safety per se. The config simply now says "use GCC *if it is available*". Similarly, `parallel` means "use parallelism *if possible*". (The doc should say that though.)

I think this new meaning is actually more useful than the previous approach, as it simplifies writing cross-platform code *using* the Linker API.